src/mlpack/bindings/markdown/print_param_table.cpp

```diff
@@ -0,0 +1,137 @@
+/**
+ * @file bindings/markdown/print_param_table.cpp
+ * @author Nippun Sharma
+ *
+ * Implementation of PrintParamTable.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core/util/io.hpp>
+#include <mlpack/core/util/binding_details.hpp>
+
+#include "binding_info.hpp"
+#include "print_param_table.hpp"
+#include "print_doc_functions.hpp"
+#include "replace_all_copy.hpp"
+
+using namespace std;
+using namespace mlpack;
+using namespace mlpack::util;
+using namespace mlpack::bindings;
+using namespace mlpack::bindings::markdown;
+
+void PrintParamTable(const string& bindingName,
+                     const string& language,
+                     Params& params,
+                     const set<string>& headers,
+                     unordered_set<string>& paramsSet,
+                     const bool onlyHyperParams,
+                     const bool onlyMatrixParams,
+                     const bool onlyInputParams,
+                     const bool onlyOutputParams)
+{
+  map<string, ParamData>& parameters = params.Parameters();
+  for (map<string, ParamData>::iterator it = parameters.begin();
+      it != parameters.end(); ++it)
+  {
+    bool printCondition = true;
+    // print output options if onlyOutputParams is true.
+    if (onlyOutputParams) printCondition &= !it->second.input;
+
+    // print input options if onlyInputParams is true.
+    if (onlyInputParams) printCondition &= it->second.input;
+
+    // print hyper parameters if onlyHyperParams is true.
+    if (onlyHyperParams)
+    {
+      bool isSerial;
+      params.functionMap[it->second.tname]["IsSerializable"](
+          it->second, NULL, (void*) &isSerial);
+
+      bool isHyperParam = false;
+      size_t foundArma = it->second.cppType.find("arma");
+      if(it->second.input && foundArma == string::npos &&
+          !isSerial)
+        isHyperParam = true;
+      printCondition &= isHyperParam;
+    }
+
+    // print matrix parameters if onlyMatrixParams is true.
+    if (onlyMatrixParams)
+    {
+      size_t foundArma = it->second.cppType.find("arma");
+      printCondition &= (foundArma != string::npos);
+    }
+
+    if (!printCondition)
+      continue;
+
+    // There are some special options that don't exist in some languages.
+    if (language != "python" && it->second.name == "copy_all_inputs")
+      continue;
+    if (language != "cli" &&
+        (it->second.name == "help" || it->second.name == "info" ||
+        it->second.name == "version"))
+      continue;
```

Comment:
I think this part may not be working right; these specific options still show up in the `Parameters` table:

http://mlpack2.ratml.org/doc/mlpack-git/python_documentation.html#adaboost
http://mlpack2.ratml.org/doc/mlpack-git/python_documentation.html#adaboost

(However, the Julia bindings seem to omit these parameters successfully?)

Reply:
Nice catch! this was due to clashing for loop variable names, fixed this :+1: